fastlane_core/lib/fastlane_core/ipa_file_analyser.rb

```diff
@@ -19,6 +19,7 @@ def self.fetch_app_version(path)
 
     # Fetches the app platform from the given ipa file.
     def self.fetch_app_platform(path)
+      path = File.expand_path(path)
```

Comment:
@ohayon I _really_ wanted to make the change in `fastlane_core/lib/fastlane_core/configuration/config_item.rb` by changing `auto_convert_value` to do this: 
```
    def auto_convert_value(value)
      return nil if value.nil?

      if data_type == Array
        return value.split(',') if value.kind_of?(String)
      elsif data_type == Integer
        return value.to_i if value.to_i.to_s == value.to_s
      elsif data_type == Float
        return value.to_f if value.to_f.to_s == value.to_s
      elsif data_type == String && allow_shell_conversion
        return value.map(&:to_s).map(&:shellescape).join(' ') if value.kind_of?(Array)
        return value.map { |k, v| "#{k.to_s.shellescape}=#{v.shellescape}" }.join(' ') if value.kind_of?(Hash)
      elsif data_type == String && value.include?("~/") # <- new case here
        return File.expand_path(value)
      else
        # Special treatment if the user specified true, false or YES, NO
        # There is no boolean type, so we just do it here
        if %w(YES yes true TRUE).include?(value)
          return true
        elsif %w(NO no false FALSE).include?(value)
          return false
        end
      end

      return value # fallback to not doing anything
    end
```
That fix broke a number of tests though that had `~/` in various paths, so I opted for the less global fix.